### ISSUE TITLE
FPS Unlocker Not Unlocking FPS

### ISSUE THREAD

**Issue by Nano-Ocelot** [Lahvuun/sekirofpsunlock#10]
Running `./sekirofpsunlock 30 set-fps 144` and nothing happens and there are no errors presented so I have no idea where to start debugging this. I have followed all instructions. Vsync disabled in dxvk file. Tried using setcap. Nothing seems to work.

**Comment by Lahvuun** [comment]
First of all, there is another solution available: using a DLL. It should, theoretically, be a much more reliable approach than mine. The project is at https://github.com/cathyjf/SekiroFpsUnlockDll and you should get specifically https://github.com/cathyjf/SekiroFpsUnlockDll/releases/download/v1.1/dinput8.dll and drop it into your game's main directory, next to `sekiro.exe`. Then set it up just as you would the Mod Engine, which happens to be described in the README here:
![image](https://github.com/Lahvuun/sekirofpsunlock/assets/6643565/5ee7171c-f6bd-44c1-8ed5-321a074c7098)
So basically just set the launch options to `WINEDLLOVERRIDES=dinput8=n,b %command%` and you should be good to go.

If you, for some reason, don't want to use that approach, or it doesn't work, then I'd like to start by taking a look at your `sekiro.exe`. Might have to compress it for GitHub to allow it to be uploaded.

**Comment by Nano-Ocelot** [comment]
Strangely this other implementation of the mod doesn't work either. I get this message in the log repeating every 3 seconds. 
`[5/13/2024 9:23:01 PM] no valid game process found...
[5/13/2024 9:23:02 PM] 	Process #0: 'Z:\home\USER\Games\SteamLibrary\steamapps\common\Sekiro\sekiro.exe' | (Z:\home\USER\Games\SteamLibrary\steamapps\common\Sekiro\sekiro.exe)
[5/13/2024 9:23:02 PM] 	Description #0: null | FromSoftware, Inc. | Sekiro™: Shadows Die Twice
[5/13/2024 9:23:02 PM] 	Data #0: 1.6.0.0 | 70066176 | 5/13/2024 9:19:07 PM | False | False`

Here is the sekiro.exe file. I couldn't get it to compress to 25MB. https://drive.google.com/file/d/1Huybutg1VibXvFlIHWicwZknW29kNP3s/view?usp=drivesdk

Some more info. I'm using OpenSUSE Tumbleweed with Plasma 6 Wayland. ProtonGE-8-32

**Comment by Lahvuun** [comment]
The fact that even the DLL is not working suggests there might be something going on with your setup. It's been years since I've played Sekiro, and it's possible that now Proton implements some sandboxing which could interfere with either way of changing the FPS.

I've requested the access to the exe on Drive, please approve it whenever you get the chance. But unless you've patched yours, it should be the default 1.6 Steam version, which would mean the problem is indeed somewhere else.

**Comment by Nano-Ocelot** [comment]
Yes it's indeed the default 1.6 exe but I've granted access

**Comment by Nano-Ocelot** [comment]
I don't know what's going on with that dll version of the unlocker but as for this one I noticed that sekiro.exe doesn't return a value when running pidof. And correct me if I'm wrong but this mod uses pidof doesn't it? It's seemingly the only game like that after testing a few games. Interestingly pgrep works just fine.

**Comment by Nano-Ocelot** [comment]
I got your mod working. It turns out it was my issue with mangohud limiting the framerate with my current settings. I never could get the other mod to work which is unfortunate because I'd really like the 'autocenter camera off' feature from that mod. Does your mod support those sorts of features as well?
